py/dml/types.py

```diff
@@ -134,6 +135,23 @@ def safe_realtype_shallow(t):
     except DMLUnknownType as e:
         raise ETYPE(e.type.declaration_site or None, e.type)
 
+def deep_const(origt):
```

Comment:
unit test?

Reply:
Beyond which I've added in `1.2/errors/ECONST`? I suppose those tests aren't very extensive, but this function is not very complicated, either.

Reply:
I was thinking of pure unit testing, something along the lines of `assert deep_const(TStruct([('foo', TInt(32, True, const=True))])`. Reduces the need to test details in T_ECONST.

Reply:
From a discussion with Erik, the conclusion was that due to the simplicity of `deep_const` and the coverage in the `ECONST` test (which would need to exist anyway) pure unit tests for `deep_const` are not absolutely necessary.